gateways/metadata_gateway.py

```
@@ -27,6 +27,18 @@ def get_dag_metadata(self, id: str) -> Optional[str]:
             return None
         return json.dumps({id: metadata})
 
+    def get_icon_metadata(self, id: str) -> Optional[bytes]:
```

Ruff (0.15.10)

[error] Function argument `id` is shadowing a Python builtin

(A002)